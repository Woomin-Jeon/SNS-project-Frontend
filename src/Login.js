/* eslint-disable react/prop-types */
/* eslint-disable max-len */
/* eslint-disable object-curly-newline */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from 'react-router-dom';
import Join from './Join';
import { closeAllChildCommentBox, closeAllPostEditBox } from './function';

const initialTempt = {
  temptId: '',
  temptPw: '',
};

function Login({
  loginState,
  setLoginState,
  currentUserState,
  setCurrentUserState,
  commentState,
  setCommentState,
  postState,
  setPostState,
}) {
  const [temptState, setTemptState] = useState(initialTempt);
  const { temptId, temptPw } = temptState;
  const { users, isLoggedIn } = loginState;
  const justTrue = true;

  const setLoginTemptId = (temptId) => {
    setTemptState({ ...temptState, temptId });
  };
  const setLoginTemptPw = (temptPw) => {
    setTemptState({ ...temptState, temptPw });
  };

  const loginButtonClicked = () => {
    for (let i = 0; i < users.length; i += 1) {
      if (temptId === users[i].id && temptPw === users[i].pw) {
        setCurrentUserState({
          ...currentUserState, id: users[i].id, pw: users[i].pw, userName: users[i].userName,
        });
        setLoginState({ ...loginState, isLoggedIn: true });
        setCommentState(closeAllChildCommentBox(commentState, justTrue));
        setPostState(closeAllPostEditBox(postState, justTrue));
      }
    }
  };

  if (isLoggedIn === true) {
    alert('로그인 성공!');
    return <Redirect to="/post" />;
  }

  return (
    <>
      <h1>Facebook 로그인 하기</h1>
      <div className="login">
        아이디 <input type="text" className="login-id-box" onChange={(e) => setLoginTemptId(e.target.value)} /><br />
        비밀번호 <input type="text" className="login-pw-box" onChange={(e) => setLoginTemptPw(e.target.value)} /><br />
        <button onClick={loginButtonClicked} className="login-button" type="button">로그인</button>
      </div>
      <br />
      <br />
      <br />
      <br />
      <Router>
        <Link to="/join" className="join-router">회원이 아니신가요?</Link>
        <Switch>
          <Route path="/join">
            <Join
              loginState={loginState}
              setLoginState={setLoginState}
            />
          </Route>
        </Switch>
      </Router>
    </>
  );
}

export default Login;
