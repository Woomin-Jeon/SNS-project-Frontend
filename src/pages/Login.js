import React, { useState, useEffect } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useHistory,
} from 'react-router-dom';
import Join from './Join';
import {
  login,
  checkSessionExist,
  getUsers,
} from '../function';
import background from '../login-background.png';

const callAPI = async (currentUserState, setCurrentUserState, loginState, setLoginState, history) => {
  const { user } = await checkSessionExist();
  const { userStore } = await getUsers();

  setCurrentUserState({
    ...currentUserState,
    id: user.id,
    userName: user.userName,
    friends: user.friends,
    profile: user.profile,
  });
  setLoginState({ ...loginState, users: [...userStore], isLoggedIn: true })

  if (user) {
    history.push('/post');
  }
};

const initialTempt = {
  temptId: '',
  temptPw: '',
};

function Login({
  loginState,
  setLoginState,
  currentUserState,
  setCurrentUserState,
  socket,
}) {
  const [temptState, setTemptState] = useState(initialTempt);
  const { temptId, temptPw } = temptState;
  const history = useHistory();

  useEffect(() => {
    callAPI(currentUserState, setCurrentUserState, loginState, setLoginState, history);
  }, []);

  const setLoginTemptId = (temptId) => {
    setTemptState({ ...temptState, temptId });
  };
  const setLoginTemptPw = (temptPw) => {
    setTemptState({ ...temptState, temptPw });
  };

  const loginButtonClicked = async () => {
    const { user, status } = await login(temptId, temptPw, socket.id);

    if (status === 400) {
      alert('아이디와 비밀번호를 다시 확인해주세요');
      return;
    }

    setCurrentUserState({
      ...currentUserState,
      id: user.id,
      userName: user.userName,
      friends: user.friends,
      profile: user.profile,
    });

    setLoginState({ ...loginState, isLoggedIn: true });
    history.push('/post');
  };

  const MoveToJoiningPage = () => {
    history.push('/join');
  };

  return (
    <>
      <div className="login-header">
        <span className="login-header-facebook">facebook</span>
        <div className="login-id">
          <div className="id-utter">아이디</div>
          <input type="text" className="login-id-box" onChange={(e) => setLoginTemptId(e.target.value)} />
        </div>
        <div className="login-pw">
          <div className="pw-utter">비밀번호</div>
          <input type="password" className="login-pw-box" onChange={(e) => setLoginTemptPw(e.target.value)} />
        </div>
        <div className="login-enter">
          <button onClick={loginButtonClicked} className="login-button" type="button">로그인</button>
        </div>
      </div>
      <Router>
        <button
          to="/join"
          className="join-router"
          onClick={MoveToJoiningPage}
        >
        회원이 아니신가요?
        </button>
        <Switch>
          <Route path="/join">
            <Join
              loginState={loginState}
              setLoginState={setLoginState}
            />
          </Route>
        </Switch>
      </Router>
      <div className="login-backgrond-utter">
        Facebook에서 전세계에 있는 친구, 가족, 지인들과 함께 이야기를 나눠 보세요
      </div>
      <img className="login-backgrond-image" src={background} />
    </>
  );
}

export default Login;
