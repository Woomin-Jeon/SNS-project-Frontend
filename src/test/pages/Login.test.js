import Login from "../../pages/Login";
import React from "react";
import {shallow} from 'enzyme';
import {BrowserRouter} from "react-router-dom";

describe('Login', () => {
  let currentUserState;
  let setCurrentUserState;
  let loginState;
  let setLoginState;
  let commentState;
  let setCommentState;
  let postState;
  let setPostState;
  let socket;

  it('renders well', () => {
    const component = shallow(
      <BrowserRouter>
        <Login
          loginState={loginState}
          setLoginState={setLoginState}
          currentUserState={currentUserState}
          setCurrentUserState={setCurrentUserState}
          commentState={commentState}
          setCommentState={setCommentState}
          postState={postState}
          setPostState={setPostState}
          socket={socket}
        />
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
