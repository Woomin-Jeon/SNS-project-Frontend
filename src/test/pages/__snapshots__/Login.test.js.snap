// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders well 1`] = `ShallowWrapper {}`;
